sheepdog/auth/__init__.py

```diff
@@ -113,9 +115,27 @@ def authorize_and_call(*args, **kwargs):
 def authorize(program, project, roles):
     resource = "/programs/{}/projects/{}".format(program, project)
     jwt = get_jwt_from_header()
-    authz = flask.current_app.auth.auth_request(
-        jwt=jwt, service="sheepdog", methods=roles, resources=[resource]
-    )
+    cache_key = str(hash((jwt, "sheepdog", tuple(roles), (resource))))
+    authz = None
+    try:
+        if AUTHZ_CACHE.has(cache_key):
+            authz = AUTHZ_CACHE.get(cache_key)
+        else:
+            authz = flask.current_app.auth.auth_request(
+                jwt=jwt, service="sheepdog", methods=roles, resources=[resource]
+            )
+            logger.info(
+                f"Retrieveing response from arborist: {authz} with {type(authz)=}"
+            )
```

Comment:
`Retrieving` typo - and i don't think this log is necessary, it would clutter the sheepdog logs. I think you're using it to debug unit tests? let's remove it when you're done